src/cephadm/cephadm

```diff
@@ -3232,10 +3232,11 @@ def command_bootstrap():
 
         logger.info('Creating initial admin user...')
         password = args.initial_dashboard_password or generate_password()
-        cmd = ['dashboard', 'ac-user-create', args.initial_dashboard_user, password, 'administrator', '--force-password']
+        tmp_password_file = write_tmp(password, uid, gid)
+        cmd = ['dashboard', 'ac-user-create', args.initial_dashboard_user, '-i', '/tmp/dashboard.pw', 'administrator', '--force-password']
         if not args.dashboard_password_noupdate:
             cmd.append('--pwd-update-required')
-        cli(cmd)
+        cli(cmd, extra_mounts={pathify(tmp_password_file.name): '/tmp/dashboard.pw:z'})
```

Comment:
I'm a bit concerned about storing the password in `/tmp` since it's readable by everyone:
```bash
# epuertat > sudo bash -c 'echo 123 > /tmp/password.txt'
# epuertat > ll /tmp/password.txt                       
-rw-r--r--. 1 root root 4 Nov 27 18:44 /tmp/password.txt
```

You mentioned that process substitution didn't work... Given it's a shell feature (bash/zsh), could it be because this is not running through a shell? I tested:
```bash
ceph dashboard ac-user-set-password test -i <(echo $PASSWORD)
```
and it worked.

And alternative, and perhaps more standard way, could be:
```bash
echo $PASSORD | ceph dashboard ac-user-set-password test -i -
```

Reply:
@jmolmo can give you more insight on why ```cephadm``` is storing password in /tmp.

Regarding process substitution, when you run it as part of ```vstart.sh``` I tested it and it does not work.

Reply:
> I'm a bit concerned about storing the password in `/tmp` since it's readable by everyone:
> 

Well, we have the tmp file "alive" while the bootstrap command is being executed, all the tmp files are removed when the bootstrap command finishes.
We are using the same strategy to pass other parameters like "admin key" to the CLI container during the bootstrap execution.
So this taking into account ... it seems very difficult that somebody can get this information that is only available in the bootstrap node during the time (1 minute or less) that the bootstrap process is being executed.

Reply:
Mmm, I think it is not really 'difficult', since just a  `cp -R /tmp/* <dst>` only requires a few seconds to expose all information. So, an unprivileged user logged into the system where cephadm is going to be deployed, what kind of information could access? Passwords, keys?

Reply:
yes, agree with you that is easy to get the information from the tmp files. What is very difficult, and i think that very likely improbable, is:
- The attacker should know what is going to be the node use to bootstrap the cluster, and gain access to this node.
- The attacker should know when the bootstrap command is going to be executed, or implement in the bootstrap node a way to collect and sen the information.

In any case , i will try to verify if your trick:

`ceph dashboard ac-user-create admin -i <(echo 'admin')  administrator --force-password`

work in cephadm

Reply:
Thank you, @jmolmo !!